### PR TITLE
Added TypeScript typings

### PR DESCRIPTION
**Pull request by TheAkio**
I inspected the objects and ended up with the following typings file. If anything should be marked as private or if something doesn't quite match up, please tell me I'll fix it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
     "name": "zlib-sync",
     "version": "0.1.5",
     "description": "Synchronous zlib bindings for NodeJS",
-    "main": "index.js",
+	"main": "index.js",
```
Comment (abalabahaha) [inline package.json]: Bad indents

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+declare module 'zlib-sync' {
+	export interface InflateOptions {
+		chunkSize?: number;
+		to?: string;
+		windowBits?: number;
+	}
+
+	export class Inflate {
+		readonly windowBits: number;
+		readonly result: Buffer;
```
Comment (abalabahaha) [inline index.d.ts]: I think this is `null`able

File: package.json (hunk 1 above)
Comment (TheAkio) [reply package.json]: Fixed that, didn't check which indentation you were using, sorry.

File: index.d.ts (hunk 2 above)
Comment (TheAkio) [reply index.d.ts]: Yup, that's nullable, also found some other places I missed typings for.

Comment (abalabahaha) [reply index.d.ts]: `| null` would be better for both of these, then

Comment (TheAkio) [reply index.d.ts]: Updated that.

**Pull request merged**